onyo/commands/new.py

```diff
@@ -42,37 +42,45 @@ def read_new_word(word_description):
     return word
 
 
-def run_onyo_new(directory, template, non_interactive, onyo_root, repo):
+def run_onyo_new(directory, template, interactive, repo):
     """
     Read a new asset name, check it for uniqueness and validity, create the
     asset (from a template if one was given), edit the new asset (if not
     suppressed with flag) and stage it with git.
 
     Returns the newly created asset's filename on success.
     """
-    # create an asset file
-    filename = create_filename(onyo_root, template)
-    filename = Path(directory, filename).resolve()
-    template = Path(template).resolve()
 
-    if filename.exists():
-        log.error(f"{filename}' asset already exists.")
-        sys.exit(1)
+    new_asset = Path(directory, create_filename(repo))
+    temp_asset = Path(repo.root, ".onyo/temp/", new_asset.name)
+
+    template = Path(template).resolve()
     if template.is_file():
-        shutil.copyfile(template, filename)
+        shutil.copyfile(template, temp_asset)
     else:
-        filename.touch()
+        log.error(f"{template} does not exist.")
```

Comment:
Nit: I like to quote user-controlled variables in single quotes, just in case something is empty or has spaces. It makes it easier debug and more obvious to the user what's going on.

Not a blocker, as `new` will be refactored soon.